### ISSUE TITLE
[ERROR] Command 'opam repository add --this-switch -k local

### ISSUE THREAD

**Issue by samoht** [tarides/ocaml-platform-installer#149]
```
* Building the tools...
  -> Creating a sandbox...
ocaml-platform: [DEBUG] Running: 'opam' 'config' 'var' 'prefix' '--yes' '-q' '--color=never' '--root' '/Users/thomas/.opam'
ocaml-platform: [INFO] Creating sandbox switch for building the tools
ocaml-platform: [DEBUG] Running: 'opam' 'switch' 'create' '--no-switch' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-sandbox-592cd5' '--empty' '--yes' '-q' '--color=never' '--root' '/Users/thomas/.opam'
ocaml-platform: [DEBUG] Running: 'opam' 'repository' 'add' '--this-switch' '-k' 'local' 'platform_sandbox_compiler_packages' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-compiler-package-ed2ff8' '--yes' '-q' '--color=never' '--switch' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-sandbox-592cd5' '--root' '/Users/thomas/.opam'
ocaml-platform: [INFO] Error in 'opam' 'repository' 'add' '--this-switch' '-k' 'local' 'platform_sandbox_compiler_packages' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-compiler-package-ed2ff8' '--yes' '-q' '--color=never' '--switch' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-sandbox-592cd5' '--root' '/Users/thomas/.opam':
[ERROR] Repository platform_sandbox_compiler_packages is already set up and points to file:///Users/thomas/.opam/plugins/ocaml-platform/platform_sandbox_compiler_packages. To change that, use 'opam repository set-url platform_sandbox_compiler_packages file:///private/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-compiler-package-ed2ff8'.
ocaml-platform: [DEBUG] Running: 'opam' 'switch' 'remove' '/var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-sandbox-592cd5' '--yes' '-q' '--color=never' '--root' '/Users/thomas/.opam'
ocaml-platform: [ERROR] Command 'opam repository add --this-switch -k local
           platform_sandbox_compiler_packages
           /var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-compiler-package-ed2ff8
           --yes -q --color=never --switch
           /var/folders/88/jkblgplx33db3c6606wyy7rc0000gn/T/ocaml-platform-sandbox-592cd5
           --root /Users/thomas/.opam' failed: exited with 2
```

**Comment by panglesd** [comment]
Thanks a lot for reporting this issue.

`opam` fails when we add a repo with the same name as an already existing one, but with a different location.
This seems to be the issue here: since #136, the sandbox compiler repo is each time in another folder but is always added with the same name (`platform_sandbox_compiler_package`).

I am however surprised such situation happen: normally, the sandbox compiler repo is removed after use. I think it can happen in few situations.

- Did you run two instances of `ocaml-platform` at the same time to obtain the error? That would then be caused by #118. I think we should have a locking mechanism, as `opam` has.
- Did you kill a previous run of `ocaml-platform` with a signal or a computer shutdown? In this case, a repository might not be cleaned (related issue: #39).

In any case, could you confirm that the output of `opam repository --all` lists a repo named `platform_sandbox_compiler_packages` when the error occurs?

**Comment by panglesd** [comment]
Fixed in #150.
